lib/main.js

```diff
@@ -78,6 +79,23 @@ const Hydrogen = {
     this.subscriptions.add(atom.workspace.observeActivePaneItem((item) => {
       if (item && item === atom.workspace.getActiveTextEditor()) {
         store.updateEditor(item);
+        store.setGrammar(item);
+      }
+    }));
+
+    this.subscriptions.add(atom.workspace.observeTextEditors((editor) => {
+      if (isMultilanguageGrammar(editor.getGrammar())) {
+        const cursorSubscription = editor.onDidChangeCursorPosition(
+          _.debounce(() => { store.setGrammar(editor); }, 75),
```

Comment:
We don't need  to switch grammar instantaneously. Therefor we debounce the event.